gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/ViewImpl.java

```diff
@@ -32,17 +44,62 @@
  */
 public abstract class ViewImpl implements View {
     private Widget widget;
+    private Map<Object, HasOneWidget> oneWidgetSlots = new HashMap<Object, HasOneWidget>();
+
+    private Map<Object,HasWidgets> hasWidgetSlots = new HashMap<Object, HasWidgets>();
 
+    private Map<OrderedSlot<?>, List<Comparable<Comparable<?>>>> orderedSlots
+            = new HashMap<OrderedSlot<?>, List<Comparable<Comparable<?>>>>();
+
+    @SuppressWarnings("unchecked")
     @Override
     public void addToSlot(Object slot, IsWidget content) {
+        if (hasWidgetSlots.containsKey(slot)) {
+            if (orderedSlots.containsKey(slot)) {
```

Comment:
What do you think not to have the slots added to both hasWidgetSlots and orderedSlots? I think it would simplify a bit (ie: `if(isInHasWidgetsSlots(slot) { /* do something */ } else if (isInOrderedSlots(slot)) { /* more stuff */ } `